### PR TITLE
[13.0][MIG] connector_magento: Migration to 13.0

### PR DESCRIPTION
**Pull request by victoralmau**
Superseed: https://github.com/OCA/connector-magento/pull/317

Migration to 13.0

New features:
- Add export_magento_qty button to force recompute qty in products.
- Add import sale order wizard by number in backend.
- Add transaction related to order.
- Add test to check import orders in different currencies.

Locked by:
- [x] `connector_ecommerce` https://github.com/OCA/connector-ecommerce/pull/62

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT30700

### CODE REVIEW

**Comment by victoralmau** [comment]
Now `connector_ecommerce` addon is already merged.
I have relaunched Travis to verify that the tests work correctly.
If you confirm that everything is OK, it could already be merged, right?

File: connector_magento/components/backend_adapter.py
Comment (joao-p-marques) [inline connector_magento/components/backend_adapter.py:230]: Don's just silence the linter in this case, as it is easy to fix:
```suggestion
            magento_api = self.work["magento_api"]
```

File: connector_magento/components/backend_adapter.py
Comment (joao-p-marques) [inline connector_magento/components/backend_adapter.py:415]: Same here as above:
```suggestion
            admin_func = self.model["_get_admin_path"]
```

File: connector_magento/models/account_invoice/common.py
Comment (joao-p-marques) [inline connector_magento/models/account_invoice/common.py:172]: I see you compare `create_invoice` with either `"paid"` or `"open"` but, a few lines above, it is asigned the value of `create_invoice_on`, which would suggest a different sort of value... Can you confirm it is being handled properly and explain a bit more why this change?

File: connector_magento/models/partner/importer.py
Comment (joao-p-marques) [inline connector_magento/models/partner/importer.py:54]:
```suggestion
```

File: connector_magento/models/partner/importer.py
Comment (joao-p-marques) [inline connector_magento/models/partner/importer.py:299]: Just to confirm, this would mean accessing `record["region"]["region"]`. Is that the expected structure?

File: connector_magento/models/partner/importer.py
Comment (joao-p-marques) [inline connector_magento/models/partner/importer.py:460]:
```suggestion
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:84]:
```suggestion
                "The order has not been paid.\nThe import will be retried later."
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:81]:
```suggestion
        amount_paid = float(record.get("payment", {}).get("amount_paid", 0.0))
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:158]:
```suggestion
        discount = float(record.get("shipping_discount_amount", 0.0))
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:160]:
```suggestion
            amount = float(record.get("base_shipping_incl_tax", 0.0)) - discount
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:162]:
```suggestion
            amount = float(record.get("shipping_amount", 0.0)) - discount
```

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:178]: You can also make the changes mentioned above to every similar occurrence (with a simple find/replace)

File: connector_magento/models/sale_order/importer.py
Comment (joao-p-marques) [inline connector_magento/models/sale_order/importer.py:872]:
```suggestion
        discount_amount = float(record.get("base_discount_amount", 0))
```

File: connector_magento/tests/magento2/common.py
Comment (joao-p-marques) [inline connector_magento/tests/magento2/common.py:46]: You can use several context managers in the same block:
```suggestion
        with mute_logger(
            "odoo.addons.mail.models.mail_mail", "odoo.models.unlink", "odoo.tests"
        ), recorder.use_cassette("metadata"):
            self.backend.synchronize_metadata()
```

File: connector_magento/tests/magento2/test_magento2_export_picking.py
Comment (joao-p-marques) [inline connector_magento/tests/magento2/test_magento2_export_picking.py:28]:
```suggestion
```

File: connector_magento/tests/magento2/test_magento2_import_partner.py
Comment (joao-p-marques) [inline connector_magento/tests/magento2/test_magento2_import_partner.py:95]: This is not the place for this sort of comments... Please check if the test is still valid and should be included, otherwise remove these lines...

File: connector_magento/tests/common.py
Comment (joao-p-marques) [inline connector_magento/tests/common.py:138]: Shouldn't you check that this actually returns what is needed and create one otherwise, like what is done above for the Cash Journal?

File: connector_magento/tests/test_export_invoice.py
Comment (joao-p-marques) [inline connector_magento/tests/test_export_invoice.py:21]: Same doubts here as above... What if the needed objects are not in the environment? You should prevent that and create the ones you need on the test...

File: connector_magento/tests/test_export_picking.py
Comment (joao-p-marques) [inline connector_magento/tests/test_export_picking.py:25]:
```suggestion
```

File: connector_magento/tests/test_import_partner.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_partner.py:96]: Same comment here as above...

File: connector_magento/tests/test_import_product_image.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_product_image.py:69]:
```suggestion
```

File: connector_magento/tests/test_import_product_image.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_product_image.py:79]:
```suggestion
```

File: connector_magento/tests/test_import_product_image.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_product_image.py:76]:
```suggestion
                    },
```

File: connector_magento/tests/test_import_product_image.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_product_image.py:86]:
```suggestion
                    },
```

File: connector_magento/tests/test_import_product_image.py
Comment (joao-p-marques) [inline connector_magento/tests/test_import_product_image.py:95]:
```suggestion
                    },
```

File: connector_magento/tests/test_sale_order.py
Comment (joao-p-marques) [inline connector_magento/tests/test_sale_order.py:33]: Same doubts here as above

**Review by joao-p-marques — CHANGES_REQUESTED** [review]
Code review

**Comment by pedrobaeza** [comment]
For now, we are holding this pull request, as we are going to speed up the transition to Odoo e-commerce, and not needing this, so o more changes should be applied, although we keep it opened as reference for anyone wanting it.

File: connector_magento/models/sale_order/importer.py
Comment (paulius-sladkevicius) [inline connector_magento/models/sale_order/importer.py:866]: We faced an issue with tax mapping as previous version a tax was set from the product settings and then a fiscal position was applied that handles correctly final taxes.

This code tries to take a tax with the same percentage even if the tax on a product is different. Extra, in case there are more than one tax (for different countries) with the same percentage it will fail as code expect to provide 1 tax, not all of them.

My suggestion is to remove this mapping and leave Odoo default behaviour when a tax is set on the SO line.

**Comment by john-herholz-dt** [comment]
Hello,

I was also working on getting the connector-magento 12.0 branch running for 15.0 and saw this pull request.
I have not yet contributed to OCA modules, but would like to do so.

At the moment I don't know where to start and who is responsible to check PRs.
I saw a lot of migrations are hanging for months because of missing migrations of other depending modules.

Long story short, who can give me a kickstart on contributing to OCA and how to communicate?

**Comment by hailangvn** [comment]
Good day @john-herholz-dt, thanks for your initiation. Hope below steps can help.
* First document to read is [`How To` section](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0#howto) to understand about how to start.
* Then in case you still want to migrate the module, you should put a comment in [tracking of migrations to 15.0](https://github.com/OCA/connector-magento/issues/328) like [this comment](https://github.com/OCA/connector-magento/issues/297#issuecomment-725314921) but with `WIP` instead of PR number. That is to inform other persons to not spend time on same work.
* While doing migration, the rest of [the guide](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0) should be read and followed.
* After creating PR, you then update the comment with the link to the PR. The early you put the link, the better, not wait until when your migration is done. That will help you receive suggestions.

**Comment by elvise** [comment]
hi @victoralmau do you have any plan to finish this task ?

**Comment by pedrobaeza** [comment]
We are not finally using it, as we switched to Odoo e-commerce, but feel free to take it from this point that is mostly OK.

**Pull request closed without merging**
